doc/manifest-tool-guide.md

```diff
@@ -0,0 +1,92 @@
+# Manifest Tool Guide
+
+This guide explains how to setup a validator so the key pairs used to sign and
+verify validations may safely change. This procedure does not require manual
+reconfiguration of servers that trust this validator.
+
+Validators use two types of key pairs: *master keys* and *ephemeral
+keys*. Ephemeral keys are used to sign and verify validations. Master keys are
+used to sign and verify manifests that change ephemeral keys. The master secret
+key should be tightly controlled. The ephemeral secret key needs to be present
+in the config file.
+
+## Validator Keys
+
+When first setting up a validator, use the `validator-keys` tool to generate a
+master key pair:
+
+```
+  $ validator-keys create_master_keys
+```
+
+Sample output:
+```
+  Master validator keys stored in /home/ubuntu/.ripple/validator-keys.json
+```
+
+Sample key file:
+```
+  {
+     "key_type" : "ed25519",
+     "master_secret" : "paUvjG1z51cpSxL3Wc9DaGTEogYk4AEesLJJaawBcVQjejxGFcb",
+     "sequence" : 4294967295,
+     "validation_public_key" : "nHUtNnLVx7odrz5dnfb2xp1gbEeJPbzJWfdicSkGyVw1eE5GpjQr"
+  }
+```
+
+The `validation_public_key` is the master public key. Any other rippled trusting
+the validator needs to add the master public key to its config. Only add keys
+received from trusted sources.
+
+The `master_secret` is the corresponding master secret key. **DO NOT SHARE THIS
+KEY**. The master secret key will be used to sign manifests that change
+ephemeral validation keys. Put the master secret key in a secure but recoverable
+location.
+
+## Validation Keys
+
+When first setting up a validator, or when changing the ephemeral keys, use the
+`validator-keys` tool to create a new ephemeral key pair:
+
+```
+  $ validator-keys create_signing_keys
```

Comment:
I've been unhappy with the words associated with building validation manifests for a while.  It predates this work.  There are too many names (e.g., ephemeral whatevers) and many of the names used in descriptions don't match what the user types or sees in the cfg file or JSON.

In an effort to clean that up a bit, I'd like to suggest a couple of renames.

- I'm wondering if we should rename the "validator_keys" command to "validation_keys".  All of the JSON and cfg stuff it produces (e.g., "validation_public_key", "validation_seed", and "validation_manifest") uses "validation" in the name.  It would be great to be consistent here.

- The name "create_master_keys" is a good one, since that's what we're actually doing, and we talk about it that way.  But I'd like to suggest that we rename the generated file from "validator-keys.json" to "validation-master-keys.json".  That way the name of the generated file aligns well with the command used to generate the file.  Also the "validation" part of the file name aligns with the "validation_public_key" inside the file.

- Suppose we rename "create_signing_keys" to "create_validation_manifest", since that's what the command produces.  Or we could rename it to "create_validation_seed" if that makes more sense to you.  Either way the command name corresponds with what the command actually produces.

- The "revoke-master-keys" command has a great name.  I don't think any changes are called for there.

These are just suggestions, not demands.  It would also be great to have @mDuo13 weigh in on this, since he'll be stuck documenting the process.